ddp-workspace/projects/ddp-sdk/src/lib/models/activity/activityTextQuestionBlock.ts

```diff
@@ -11,12 +11,27 @@ export class ActivityTextQuestionBlock extends ActivityQuestionBlock<string> {
     public confirmEntry: boolean;
     public confirmPrompt: string | null = null;
     public mismatchMessage: string | null = null;
+    public confirmationValue: string | null;
 
     constructor() {
-        super();
+      super();
     }
 
     public get questionType(): QuestionType {
-        return QuestionType.Text;
+      return QuestionType.Text;
+    }
+
+    public get answer(): string | null {
+        return this._answer;
+    }
+
+    public set answer(value: string | null) {
+        this.setAnswer(value);
+    }
+
+    setAnswer(value: string | null, doValidation: boolean = true): void {
+      this._answer = value;
+      this.confirmEntry && (this.confirmationValue = value);
+      doValidation && this.validate();
     }
```

Comment:
Marco, I want to discuss this logic...
The main idea that I wanted to show in my implementation is that blocks shouldn't know about the confirmation filed.
Email input type no different from other components - there should store only one answer, so, can save the email input type answer in the block.answer only if form valid, we shouldn't store both answers. Block shouldn't validate answers, we can delegate it to Angular form in the email input type component.
I am inclined to believe that we should remove new logic in the `ActivityQuestionBlock`, because we mixed validation logic and storing answers; remove `ActivityEmailValidatorRule`, because we can delegate validation to Angular form; remove this logic https://github.com/broadinstitute/ddp-angular/blob/f3372ae9cffe9960ccd9594a234e37e5160a6c6a/ddp-workspace/projects/ddp-sdk/src/lib/services/activity/activityQuestionConverter.service.ts#L66 because we delegate validation to Angular form.
My idea is - we don't need external validation rules, Angular form in the email input type component can validate all the stuff and store only valid email address.

Reply:
Thanks for taking a look during vacation. Hopefully you still taking time to relax.
We can talk further about this topic on Monday, but just know that I am not adding this without careful consideration.

My #1 least favorite thing in our system is how we do validations (stateful, triggered multiple times and sometimes by setting a flag, etc.etc) . But as things stand right now, validation error messages displayed on activity instance pages ("wizard" and "regular") are generated by validator rules that are attached to question "blocks".

The email validator rule could of course be added in some other way than what I have done here, but essentially same functionality needs to be added somewhere.

I like what you did with FormGroup validator (and left it in there mostly unchanged), but this validation would not prevent the final activity PUT of the activity or display an error message similar to what we have in the other question components (i.e., <ddp-validation-message>).

At any rate, don't worry about this now. Let's talk Monday. We should talk about just validations in general too, which as I said in the beginning, not really happy with.

Reply:
Don't worry, I had a few munites to take a look at PR, but I swear, I won't open GitHub till the end of the vacation 😄

> but this validation would not prevent the final activity PUT of the activity 

I guess the REQUIRED validation rule should prevent it.
Moreover, for backend valid answers for email input type are strings and null, my code
```TS
    public onChange(answer: string): void {
        const allFieldsTouched = Object.values(this.emailForm.controls).every(control => control.touched);
        if (this.emailForm.valid) {
            this.saveAnswer(answer);
        } else if (allFieldsTouched) {
            this.saveAnswer(null);
        }
    }
```
satisfied these rules.

But okay, let's discuss it on the next week, not on Monday, on Thursday, because we're coming back from vacation on Thursday.

Reply:
So, since I'm back, let's return to the discussion 🙂
My strong opinion - right now we shouldn't change validation behavior for one block. If we want to do so, we need to discuss new validation architecture in general. 

So, talking about email input type, your changes suggest that if emails mismatched or a user inputted invalid email, the block becomes invalid for activity. But as I mentioned above, it implicitly does question required and it can affect UX when a question shouldn't be required. Email input type logic should be: if email input form valid - the user answered the question if invalid - the user didn't answer the question. The REQUIRED validation rule will control that a user answered on this question.